src/public/js/util.js

```diff
@@ -257,7 +265,7 @@ var UTIL = (function() {
       }
     }
 
-    return remainder;
+    return trimLeadingZeros(remainder);
```

Comment:
As is, this will break the encode and decode functions of RS. I told them to expect a remainder of length n-k. That could be refactored though. I think it would be fine to leave the remainder untrimmed since we expect the remainder to always be a certain length and we often expect it to return all 0's. especially on decoding. Up to you. It would be easy to refactor the RS encode and decode to work around trimmed 0s here.